src/connection-manager/connections.ts

```diff
@@ -100,129 +90,87 @@ async function setHandlers(
         return
       }
 
-      if (data.result?.node) {
-        void handleWsMessageLedgerEntryAmendments(
-          data as LedgerEntryResponse,
-          networks,
-        )
-      } else if (data.result?.ledger && isInitialNode) {
-        void handleWsMessageLedgerEnableAmendments(
-          data as LedgerResponseExpanded,
-          networks,
-        )
-      } else {
-        void handleWsMessageSubscribeTypes(
-          data,
-          ledger_hashes,
-          networks,
-          networkFee,
-          ws,
-          validationNetworkDb,
-        )
-      }
+      void handleWsMessageSubscribeTypes(
+        data,
+        ledger_hashes,
+        network,
+        networkFee,
+        ws,
+        validationNetworkDb,
+        enableAmendmentLedgerIndexMap,
+      )
     })
     ws.on('close', async (code, reason) => {
       log.error(
-        `Websocket closed for ${ws.url} on ${
-          networks ?? 'unknown network'
-        } with code ${code} and reason ${reason.toString('utf-8')}.`,
+        `Websocket closed for ${
+          ws.url
+        } on ${network} with code ${code} and reason ${reason.toString(
+          'utf-8',
+        )}.`,
       )
 
+      void updateConnectionHealthStatus(ws.url, false)
+      ws.terminate()
+
       const delay = BASE_RETRY_DELAY * 2 ** retryCount
 
       if (CLOSING_CODES.includes(code) && delay <= MAX_RETRY_DELAY) {
-        log.info(
-          `Reconnecting to ${ws.url} on ${
-            networks ?? 'unknown network'
-          } after ${delay}ms...`,
-        )
-        // Clean up the old Websocket connection
-        connections.delete(ws.url)
-        ws.terminate()
-        resolve()
+        log.info(`Reconnecting to ${ws.url} on ${network} after ${delay}ms...`)
 
         setTimeout(async () => {
           // Open a new Websocket connection for the same url
           const newWS = new WebSocket(ws.url, { handshakeTimeout: WS_TIMEOUT })
 
-          await setHandlers(
-            ws_url,
-            newWS,
-            networks,
-            isInitialNode,
-            retryCount + 1,
-          )
+          await setHandlers(newWS, publicKey, network, retryCount + 1)
         }, delay)
-
-        // return since the old websocket connection has already been terminated
-        return
       }
 
-      if (connections.get(ws.url)?.url === ws.url) {
-        connections.delete(ws.url)
-        void saveNodeWsUrl(ws.url, false)
-      }
-      ws.terminate()
       resolve()
     })
     ws.on('error', (err) => {
       log.error(
-        `Websocket connection error for ${ws.url} on ${
-          networks ?? 'unknown network'
-        } - ${err.message}`,
+        `Websocket connection error for ${ws.url} on ${network} - ${err.message}`,
```

Comment:
Same here. Since we try multiple ports for an url, this will raise lots of errors that has no important implication